transport/websocketpeer.go

```diff
@@ -115,7 +116,8 @@ func ConnectWebsocketPeer(routerURL string, serialization serialize.Serializatio
 		dialer.EnableCompression = true
 	}
 
-	conn, _, err := dialer.Dial(routerURL, nil)
+	ctx, _ := context.WithTimeout(context.Background(), 5 * time.Second)
```

Comment:
The approach that I would prefer for this is for the caller to pass in a context. That way the caller can set whatever timeout/deadline they want, or call the cancel function stop waiting.

With this code above, it would still be good to defer-call the returned cancel function so that resources are freed as soon as `ConnectWebSocketPeer()` exits

Reply:
Agreed, this would be nice to pass a context but would break backwards compatibility. How about a ConnectWebsocketPeerContext function that can pass a context, similar to Dial and DialContext? Then I can refactor the ConnectWebsocketPeer to simply call ConnectWebsocketPeerContext with a nil context

Reply:
Yes, that is what I was looking for.  I would like both a `ConnectWebsocketPeerContext` and a `ConnectRawSocketPeerContext`.  If those are provided, then the above code in `ConnectWebsocketPeer` would change to to create a context and call the new `ConnectWebsocketPeerContext` (can defer-call the cancel func).

I will wait for the PR update to introduce those changes before merging anything.